drivers/dma/dma_tt_bh_noc.c

```diff
@@ -0,0 +1,711 @@
+	}
+
+	/* Cancel any pending async work for this channel */
+	k_work_cancel_delayable(&chan_data->work_ctx.work);
+
+	/* Stop cyclic transfers */
+	chan_data->cyclic_active = false;
+	chan_data->suspended = false;
+	chan_data->active = false;
+	chan_data->block_index = 0;
+
+	return 0;
+}
+
+static int noc_dma_suspend(const struct device *dev, uint32_t channel)
+{
+	struct tt_bh_dma_channel_data *chan_data = get_channel_data(dev, channel);
+
+	if (!chan_data) {
+		LOG_ERR("Invalid channel %u", channel);
+		return -EINVAL;
+	}
+
+	if (!chan_data->active) {
+		LOG_ERR("Channel %u not active", channel);
+		return -EINVAL;
+	}
+
+	chan_data->suspended = true;
+	LOG_DBG("Suspended channel %u", channel);
+
+	return 0;
+}
+
+static int noc_dma_resume(const struct device *dev, uint32_t channel)
+{
+	struct tt_bh_dma_channel_data *chan_data = get_channel_data(dev, channel);
+
+	if (!chan_data) {
+		LOG_ERR("Invalid channel %u", channel);
+		return -EINVAL;
+	}
+
+	if (!chan_data->active) {
+		LOG_ERR("Channel %u not active", channel);
+		return -EINVAL;
+	}
+
+	chan_data->suspended = false;
+	LOG_DBG("Resumed channel %u", channel);
+
+	/* If cyclic and suspended, reschedule work */
+	if (chan_data->config.cyclic && chan_data->cyclic_active) {
+		k_work_reschedule(&chan_data->work_ctx.work, K_NO_WAIT);
+	}
+
+	return 0;
+}
+
+static void noc_dma_release_channel(const struct device *dev, uint32_t channel)
+{
+	struct tt_bh_dma_noc_data *data = (struct tt_bh_dma_noc_data *)dev->data;
+	struct tt_bh_dma_channel_data *chan_data = get_channel_data(dev, channel);
+
+	if (!chan_data) {
+		LOG_ERR("Invalid channel %u", channel);
+		return;
+	}
+
+	k_mutex_lock(&data->lock, K_FOREVER);
+
+	/* Stop any ongoing transfer */
+	if (chan_data->active) {
+		noc_dma_stop(dev, channel);
+	}
+
+	/* Reset channel state */
+	chan_data->configured = false;
+	chan_data->active = false;
+	chan_data->suspended = false;
+	chan_data->cyclic_active = false;
+	chan_data->transfer_count = 0;
+	chan_data->block_index = 0;
+	chan_data->block_count = 0;
+	/* Clear blocks array */
+	memset(chan_data->blocks, 0, sizeof(chan_data->blocks));
+
+	/* Mark channel as free (safe even if not previously allocated) */
+	data->channel_allocator &= ~BIT(channel);
+
+	k_mutex_unlock(&data->lock);
+}
+
+static const struct dma_driver_api noc_dma_api = {
+	.config = noc_dma_config,
+	.reload = NULL,
+	.start = noc_dma_start,
+	.stop = noc_dma_stop,
+	.suspend = noc_dma_suspend,
+	.resume = noc_dma_resume,
+	.get_status = NULL,
+	.get_attribute = NULL,
+	.chan_filter = NULL,
+	.chan_release = noc_dma_release_channel,
+};
+
+#define NOC_DMA_INIT(n)                                                                            \
+	static const struct tt_bh_dma_noc_config noc_dma_config_##n = {                            \
+		.max_channels = CONFIG_DMA_TT_NOC_MAX_CHANNELS,                                    \
+	};                                                                                         \
```

Comment:
https://github.com/zephyrproject-rtos/zephyr/blob/545d40e3141ea73ac871b81d943e4b2cad5fe3b2/dts/bindings/dma/dma-controller.yaml#L23

```suggestion
	struct tt_bh_dma_channel_data noc_dma_config_##n_channels[DT_INST_PROP_GET(n, dma_channels)]; \
	static const struct tt_bh_dma_noc_config noc_dma_config_##n = {                            \
	    .channels = noc_dma_config_##n_channels, \
		.max_channels = CONFIG_DMA_TT_NOC_MAX_CHANNELS,                                    \
	};                                                                                         \
```